openedx/core/djangoapps/content/block_structure/tasks.py

```diff
@@ -15,9 +15,17 @@
     default_retry_delay=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_DEFAULT_RETRY_DELAY'],
     max_retries=settings.BLOCK_STRUCTURES_SETTINGS['BLOCK_STRUCTURES_TASK_MAX_RETRIES'],
 )
-def update_course_in_cache(course_key):
+def update_course_in_cache(course_id):
     """
     Updates the course blocks (in the database) for the specified course.
     """
-    course_key = CourseKey.from_string(course_key)
-    api.update_course_in_cache(course_key)
+    try:
+        course_key = CourseKey.from_string(course_id)
+        api.update_course_in_cache(course_key)
+    except Exception as exc:   # pylint: disable=broad-except
+        if isinstance(exc, NotImplementedError):
```

Comment:
What happens if this function is called with an ID for a course that does not exist? Will that throw a `DoesNotExist`? I would imagine that's a guaranteed failure as well, right?